src/cargo/core/resolver/encode.rs

```diff
@@ -17,6 +17,12 @@ pub struct EncodableResolve {
     /// `root` is optional to allow forward compatibility.
     root: Option<EncodableDependency>,
     metadata: Option<Metadata>,
+    augment: Option<Augment>,
```

Comment:
I think we can use null object pattern and store `augment: Augment`, using an empty augment instead of `None`. Serialization can be tweaked separately:

```
#[serde(default)]
#[serde(skip_if = "Augment::is_empty")]
augment: Augment
```